### PR TITLE
Adding a custom validation for flexible VM naming

### PR DESCRIPTION
**Pull request by maayarosama**
### Description

Added a custom validation decorator accepting alphanumeric values and underscores

### Related Issues

- #2303
### Documentation PR

For UI changes, Please provide the Documetation PR on [info_grid](https://github.com/threefoldtech/info_grid)

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/grid_client/src/helpers/validator.ts [hunk 1]
```diff
@@ -31,4 +31,25 @@ function validateHexSeed(seed: string, length: number): boolean {
   return true;
 }
 
-export { validateObject, validateInput, validateHexSeed };
+function IsAlphanumericExceptUnderscore(validationOptions?: ValidationOptions) {
```
Comment (0oM4R) [inline packages/grid_client/src/helpers/validator.ts:34]: is underscore allowed or not? :D

**Review by Mahmoud-Emad — APPROVED** [review]

**Review by amiraabouhadid — COMMENTED** [review]
it isnt accepting underscore 
image

**Review by 0oM4R — CHANGES_REQUESTED** [review]
` BaseGetDeleteModel` need to be updated as well

**Review by 0oM4R — COMMENTED** [review]
also `BaseGetDeleteModel` need to be updated
image

File: packages/playground/src/components/caprover_worker.vue [hunk 2]
```diff
@@ -4,10 +4,10 @@
       :rules="[
         validators.required('Name is required.'),
         validators.isLowercase('Name should consist of lowercase letters only.'),
-        validators.isAlphanumeric('Name should consist of letters and numbers only.'),
+        validators.IsAlphanumericExpectUnderscore('Name should consist of letters ,numbers and undersocres only.'),
```
Comment (amiraabouhadid) [inline packages/playground/src/components/caprover_worker.vue:7]: please change typo to "underscores".

**Review by amiraabouhadid — CHANGES_REQUESTED** [review]

File: packages/playground/src/weblets/freeflow.vue [hunk 3]
```diff
@@ -18,9 +18,9 @@
         :value="threebotName"
         :rules="[
           validators.required('Name is required.'),
-          validators.isAlphanumeric('Name should consist of letters and numbers only.'),
+          validators.IsAlphanumericExpectUnderscore('Name should consist of letters ,numbers and undersocres only.'),
```
Comment (amiraabouhadid) [inline packages/playground/src/weblets/freeflow.vue:21]: here too

**Comment by Mahmoud-Emad** [comment]
Please fix the conflicts

**Review by amiraabouhadid — APPROVED** [review]

**Review by zaelgohary — COMMENTED** [review]
I think the maxLength should be changed here:

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/8ea81fe95aa870ec9eca4ba72f8d84033191f626/packages/grid_client/src/modules/models.ts#L344

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/8ea81fe95aa870ec9eca4ba72f8d84033191f626/packages/grid_client/src/modules/models.ts#L237

File: packages/playground/src/weblets/tf_presearch.vue [hunk 4]
```diff
@@ -29,11 +29,10 @@
           :value="name"
           :rules="[
             validators.required('Name is required.'),
-            validators.isLowercase('Name should consist of lowercase letters only.'),
             validators.isAlphanumeric('Name should consist of letters and numbers only.'),
```
Comment (zaelgohary) [inline packages/playground/src/weblets/tf_presearch.vue:32]: should be `IsAlphanumericExpectUnderscore`.

File: packages/playground/src/weblets/tf_subsquid.vue [hunk 5]
```diff
@@ -17,11 +17,10 @@
         :value="name"
         :rules="[
           validators.required('Name is required.'),
-          validators.isLowercase('Name should consist of lowercase letters only.'),
           validators.isAlphanumeric('Name should consist of letters and numbers only.'),
```
Comment (zaelgohary) [inline packages/playground/src/weblets/tf_subsquid.vue:20]: here too.

**Review by zaelgohary — APPROVED** [review]

**Pull request merged**